api/provider/aws/environment_scaler.go

```diff
@@ -1,14 +1,629 @@
+	}
+
+	return consumerModels
+}
+
+func resourceProviderModels(providers []*models.ResourceProvider) []models.ResourceProvider {
+	providerModels := make([]models.ResourceProvider, len(providers))
+	for i, provider := range providers {
+		newProvider := models.ResourceProvider{
+			ID:              provider.ID,
+			InUse:           provider.InUse,
+			UsedPorts:       provider.UsedPorts,
+			AvailableMemory: provider.AvailableMemory,
+		}
+		providerModels[i] = newProvider
+	}
+
+	return providerModels
+}
+
+func (e *EnvironmentScaler) getResourceProviders(clusterName string) ([]*models.ResourceProvider, error) {
+	listContainerInstancesInput := &ecs.ListContainerInstancesInput{}
+	listContainerInstancesInput.SetCluster(clusterName)
+
+	containerInstanceARNs := []*string{}
+	listContainerInstancesPagesFN := func(output *ecs.ListContainerInstancesOutput, lastPage bool) bool {
+		containerInstanceARNs = append(containerInstanceARNs, output.ContainerInstanceArns...)
+
+		return !lastPage
+	}
+
+	if err := e.Client.ECS.ListContainerInstancesPages(listContainerInstancesInput, listContainerInstancesPagesFN); err != nil {
+		return nil, err
+	}
+
+	describeContainerInstancesInput := &ecs.DescribeContainerInstancesInput{}
+	describeContainerInstancesInput.SetCluster(clusterName)
+	describeContainerInstancesInput.SetContainerInstances(containerInstanceARNs)
+
+	output, err := e.Client.ECS.DescribeContainerInstances(describeContainerInstancesInput)
+	if err != nil {
+		return nil, err
+	}
+
+	result := []*models.ResourceProvider{}
+	if len(output.ContainerInstances) == 0 {
+		return result, nil
+	}
+
+	for _, instance := range output.ContainerInstances {
+		if !aws.BoolValue(instance.AgentConnected) {
+			continue
+		}
+
+		if aws.StringValue(instance.Status) != "ACTIVE" {
+			continue
+		}
+
+		// it's non-intuitive, but the ports being used by the tasks live in
+		// instance.RemainingResources, not instance.RegisteredResources
+		var (
+			usedPorts       []int
+			availableMemory bytesize.Bytesize
+		)
+
+		for _, resource := range instance.RemainingResources {
+			switch aws.StringValue(resource.Name) {
+			case "MEMORY":
+				val := aws.Int64Value(resource.IntegerValue)
+				availableMemory = bytesize.MiB * bytesize.Bytesize(val)
+
+			case "PORTS":
+				for _, port := range resource.StringSetValue {
+					port, err := strconv.Atoi(aws.StringValue(port))
+					if err != nil {
+						log.Printf("[WARN] Instance %s: Failed to convert port to int: %v", aws.StringValue(instance.Ec2InstanceId), err)
+						continue
+					}
+
+					usedPorts = append(usedPorts, port)
+				}
+			}
+		}
+
+		inUse := aws.Int64Value(instance.PendingTasksCount)+aws.Int64Value(instance.RunningTasksCount) > 0
+		r := &models.ResourceProvider{
+			ID:              aws.StringValue(instance.Ec2InstanceId),
+			InUse:           inUse,
+			UsedPorts:       usedPorts,
+			AvailableMemory: fmt.Sprintf("%v", availableMemory), //todo: this is definitely not correct
```

Comment:
Assuming that the `AvailableMemory` field here accepts a string that is a numeric value and the unit (for instance, 10MB), could you not leverage the [Format()](https://github.com/zpatrick/go-bytesize/blob/master/bytesize.go#L112) method in go-bytesize?

Reply:
Good catch, It looks like we could.